### ISSUE TITLE
thenAnswerMany

### ISSUE THREAD

**Issue by kmcgill88** [shinayser/mocktailx#5]
I frequently run across the use case of needing to return multiple different responses based on invocation n, n+1. I found this snippet and thought it might be a good fit for this package. If you agree I'm happy to PR :)  

```
extension WhenExtension<T> on When<T> {
  thenAnswerMany(List<Answer<T>> cbs) {
    return thenAnswer((invocation) {
      if (cbs.isEmpty) throw "No more answers available";
      return cbs.removeAt(0)(invocation);
    });
  }
}
```
then to use 

```
when(() => mock.call()).thenAnswerMany([
  (invocation) => result1,
  (invocation) => result2,
]);
```
Reference [stackoverflow](https://stackoverflow.com/questions/69687714/mock-different-answers-and-exceptions-after-several-calls-with-mocktail)

**Comment by shinayser** [comment]
Hey man! Sounds pretty good idea. I would only change the name from thenAnswerMany to thenAnswerInRow to be more expressive.

Can you submit a PR?

**Comment by shinayser** [comment]
Its published on 0.0.6!  https://pub.dev/packages/mocktailx
Thanks for your work @kmcgill88 .
